hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/InvocationTimeoutTest.java

```diff
@@ -217,4 +231,73 @@ private boolean isInterruptionCaught() {
         protected abstract void doOp() throws InterruptedException;
     }
 
+    @Test
+    public void test_operationExecution_whenOperationTimedOut() {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance hz = factory.newHazelcastInstance();
+        factory.newHazelcastInstance();
+
+        warmUpPartitions(factory.getAllHazelcastInstances());
+
+        NodeEngineImpl nodeEngine = getNodeEngineImpl(hz);
+
+        String key = randomString();
+        int partitionId = nodeEngine.getPartitionService().getPartitionId(key);
+
+        long callTimeout = 3000L;
+        LongRunningOperation longRunningOperation = new LongRunningOperation(2 * callTimeout + 1000);
+        InternalOperationService operationService = nodeEngine.getOperationService();
+        operationService.invokeOnPartition(null, longRunningOperation, partitionId);
+
+        InternalCompletableFuture<Object> future = operationService.createInvocationBuilder(null,
+                new LockOperation(new InternalLockNamespace(key), nodeEngine.toData(key), 1, -1), partitionId)
+                .setCallTimeout(callTimeout).invoke();
+
+        boolean timedOut;
```

Comment:
It should be `OperationTimeoutException` but I don't remember why other path is there :) Probably  a left over from when I was playing with different timeouts scenarios. Removed the other path.